File: src/Backend/Domain/Auth.Entity/Ids4Entity/ApiResource.cs
using System;
using Dapper.Contrib.Plus;

namespace Auth.Entity.Ids4Entity
{
    /// <summary>
    /// ApiResource
    /// </summary>
    [Table("api_resources")]
    public class ApiResource
    {
        /// <summary>
        /// 主键
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Enabled
        /// </summary>
        public bool Enabled { get; set; }

        /// <summary>
        /// Name
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// DisplayName
        /// </summary>
        public string DisplayName { get; set; }

        /// <summary>
        /// Description
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// AllowedAccessT
        /// </summary>
        public string AllowedAccessT { get; set; }

        /// <summary>
        /// ShowInDiscover
        /// </summary>
        public bool ShowInDiscover { get; set; }

        /// <summary>
        /// Created
        /// </summary>
        public string Created { get; set; }

        /// <summary>
        /// Updated
        /// </summary>
        public string Updated { get; set; }

        /// <summary>
        /// LastAccessed
        /// </summary>
        public string LastAccessed { get; set; }

        /// <summary>
        /// NonEditable
        /// </summary>
        public bool NonEditable { get; set; }
    }
}

File: src/FrontEnd/auth-admin/src/types/entities/ids/api-resource.ts
interface IApiResource {
  id?: string;
  enabled: boolean;
  name: string;
  displayName: string;
  description: string;
  allowedAccessT: string;
  showInDiscover: boolean;
  created: string;
  updated: string;
  lastAccessed: string;
  nonEditable: boolean;
}

export default class ApiResource implements IApiResource {
  id?: string;
  enabled: boolean = true;
  name: string = '';
  displayName: string = '';
  description: string = '';
  allowedAccessT: string = '';
  showInDiscover: boolean = false;
  created: string = '';
  updated: string = '';
  lastAccessed: string = '';
  nonEditable: boolean = false;
}